h/api/storage.py

```diff
@@ -50,19 +49,37 @@ def fetch_annotation(session, id_):
         return None
 
 
-def fetch_ordered_annotations(session, ids, load_documents=False):
+def fetch_ordered_annotations(session, ids, query_processor=None):
+    """
+    Fetch all annotations with the given ids and order them based on the list
+    of ids.
+
+    The optional `query_processor` parameter allows for passing in a function
+    that can change the query before it is run, especially useful for
+    eager-loading certain data. The function will get the query as an argument
+    and has to return a query object again.
+
+    :param session: the database session
+    :type session: sqlalchemy.orm.session.Session
+
+    :param ids: the list of annotation ids
+    :type id_: list
+
+    :param query_processor: an optional function that takes the query and
+                            returns an updated query
+    :type query_processor: function
```

Comment:
You're using query_processor as a callable here (i.e. not specifically as a function), so I'd be inclined to document it as a callable, not a function.